File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/cheats/model/GraphicsModGroup.java
package org.dolphinemu.dolphinemu.features.cheats.model;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

public class GraphicsModGroup
{
  @Keep
  private final long mPointer;

  @Keep
  private GraphicsModGroup(long pointer)
  {
    mPointer = pointer;
  }

  @Override
  public native void finalize();

  @NonNull
  public native GraphicsMod[] getMods();

  public native void save();

  @NonNull
  public static native GraphicsModGroup load(String gameId);
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/cheats/model/GraphicsModGroup.kt
package org.dolphinemu.dolphinemu.features.cheats.model

import androidx.annotation.Keep

class GraphicsModGroup @Keep private constructor(@field:Keep private val pointer: Long) {
    external fun finalize()

    val mods: Array<GraphicsMod>
        external get

    external fun save()

    companion object {
        @JvmStatic
        external fun load(gameId: String): GraphicsModGroup
    }
}
